apps/api/src/app/bridge/bridge.controller.ts

```diff
@@ -174,12 +174,17 @@ export class BridgeController {
     if (!workflowExist) {
       throw new NotFoundException('Workflow not found');
     }
+    const step = workflowExist?.steps.find((item) => item.stepId === stepId);
+
+    if (!step || !step._id) {
+      throw new NotFoundException('Step not found');
+    }
```

Comment:
Question: can we do something like this?
I started to use this pattern in [this PR](https://github.com/novuhq/novu/pull/6651/files#diff-b92f8bced0d347f2344f3113183234889caa44ab84492116e7c4edc4e6cc8ef8) this way we have the clean message to search by with the additional metadata (step id) 

```suggestion
    if (!step || !step._id) {
      throw new NotFoundException({
        message: 'Step not found',
        stepId: stepId,
      });
    }
```

Reply:
As discussed, we will address this holistically across the API using shared classes.